### PR TITLE
Added coverage report step

### PR DESCRIPTION
**Pull request by LukasHirt**

### CODE REVIEW

**Comment by LukasHirt** [comment]
@micbar I think this should be able to report the coverage to codacy. Can you pls check if the code makes sense and also add the token as a secret?

File: .drone.star [hunk 1]
```diff
@@ -56,6 +56,20 @@ def testing(ctx):
                     'yarn test',
                 ]
             },
+            {
+                'name': 'report coverage to codacy',
+                'image': 'ubuntu',
+                'pull': 'always',
```
Comment (micbar) [inline .drone.star:62]: ubuntu? why not owncloudci/alpine? much smaller

File: .drone.star [hunk 2]
```diff
@@ -56,6 +56,20 @@ def testing(ctx):
...
+                'commands': [
+                    'export CODACY_PROJECT_TOKEN=$CODACY_TOKEN',
+                    '/bin/bash <(curl -Ls https://coverage.codacy.com/get.sh)',
+                ],
+                'environment': {
+                    'CODACY_TOKEN': {
+                        'from_secret': 'codacy_token',
```
Comment (micbar) [inline .drone.star:69]: did you try to set `CODACY_PROJECT_TOKEN` directly

File: .drone.star [hunk 3]
```diff
@@ -56,6 +56,20 @@ def testing(ctx):
                     'yarn test',
                 ]
             },
+            {
+                'name': 'report coverage to codacy',
```
Comment (micbar) [inline .drone.star:60]: `codacy` would nbe sufficient, alignment with the other pipelines

File: .drone.star (hunk 2 above)
Comment (LukasHirt) [reply .drone.star:69]: I thought the token shouldn't be directly exposed in the code. But if it's not a problem I'll do it like that 😉

Comment (micbar) [reply .drone.star]: just the variable name, the value is not accessible -> secrets

Comment (LukasHirt) [reply .drone.star]: So skipping the export command and have it only as env?

Comment (micbar) [reply .drone.star]: But we agree to hide the token value in the secrets.

Comment (micbar) [reply .drone.star]: is now in the secrets

Comment (micbar) [reply .drone.star]: we need to reset it after the successful test

**Comment by LukasHirt** [comment]
Getting error: `Exception in thread "main" java.net.UnknownHostException: api.codacy.com` now 😞

**Comment by micbar** [comment]
> Getting error: `Exception in thread "main" java.net.UnknownHostException: api.codacy.com` now 😞

You now reached the same stage than me 2 weeks ago. 😄

**Comment by LukasHirt** [comment]
Adding command `echo "$(dig +short api.codacy.com | tail -n1) api.codacy.com" >> /etc/hosts` as described [here](https://github.com/codacy/codacy-coverage-reporter/blob/master/README.md#substratesegfaulthandler-caught-signal-11) seems to help with the error but there are now two new ones.

This, however, requires getting `bind-tools` package before that step which is ugly 🤢 Otherwise we are getting an error about missing `dig` command.

**Comment by LukasHirt** [comment]
> Adding command echo "$(dig +short api.codacy.com | tail -n1) api.codacy.com" >> /etc/hosts as described here  seems to help with the error but there are now two new ones.

Ok, now it fails again with the same error. So it doesn't seem to help.

**Comment by LukasHirt** [comment]
I tried to use now the official [alpine image](https://hub.docker.com/_/alpine) and the result seems to be the same @micbar

**Comment by micbar** [comment]
@LukasHirt got an answer in the ticket at codacy. Now using an alpine image with glibc as an intermediate solution.

**Comment by LukasHirt** [comment]
> got an answer in the ticket at codacy. Now using an alpine image with glibc as an intermediate solution.

Awesome! Let's squash and merge then? @micbar

**Pull request merged**